### ISSUE TITLE
Comparing string with number

### ISSUE THREAD

**Issue by gendi** [Jaliborc/Levelland#1]
Hello

When accepting quests from Koda Steelclaw (http://www.wowhead.com/npc=91223), the quest "Out of the Dream" returns -1 at line 60 for `local level = select((i - start) * n + 2, ...)`, but "The Demons Below" returns a string of "The Demons Below" at line 60 and creates an error at line 62.

I haven't looked into what line 60 is trying to get, but instead a quick workaround for the error until then was to add `type(level) == number and ` to the if statement at line 62

**Comment by gendi** [comment]
Still an issue with with May 11 release

**Comment by ZaephirMoth** [comment]
Greetings !

  I've just started to use Levelland and encountered several lua errors similar to the one described above. It was while talking to the NPC Darkcaller Yanka (during Hallow's End), which brings one two daily quests ("A time to build up" and "A time to break down"). But it happened again, shortly after, with NPC Candace Fenlow (another two daily quests : "Clean up Undercity" and "Stink bombs away"), though it didn't occur with NPC Spoops located near her (non-daily/non-repeatable quest : "Hallow's End Treats for Spoops"), so I guess this bug does only concern daily quests. Making progress in those quests seems to also cause troubles...

  1) Message: Interface\AddOns\Levelland\Levelland.lua:62: attempt to compare number with string
Time: 10/24/17 01:01:20
Count: 1
Stack: Interface\AddOns\Levelland\Levelland.lua:62: attempt to compare number with string
[C]: ?
Interface\AddOns\Levelland\Levelland.lua:62: in function <Interface\AddOns\Levelland\Levelland.lua:56>
[C]: in function `GossipFrameAvailableQuestsUpdate'
Interface\FrameXML\GossipFrame.lua:40: in function `GossipFrameUpdate'
Interface\FrameXML\GossipFrame.lua:29: in function <Interface\FrameXML\GossipFrame.lua:10>

Locals: 

  2) Message: Interface\AddOns\Levelland\Levelland.lua:62: attempt to compare number with string
Time: 10/24/17 01:20:57
Count: 1
Stack: Interface\AddOns\Levelland\Levelland.lua:62: attempt to compare number with string
[C]: ?
Interface\AddOns\Levelland\Levelland.lua:62: in function <Interface\AddOns\Levelland\Levelland.lua:56>
[C]: in function `GossipFrameActiveQuestsUpdate'
Interface\FrameXML\GossipFrame.lua:41: in function `GossipFrameUpdate'
Interface\FrameXML\GossipFrame.lua:29: in function <Interface\FrameXML\GossipFrame.lua:10>

Locals: 

  3) Message: Interface\AddOns\Levelland\Levelland.lua:62: attempt to compare number with string
Time: 10/24/17 01:26:40
Count: 96
Stack: Interface\AddOns\Levelland\Levelland.lua:62: attempt to compare number with string
[C]: ?
Interface\AddOns\Levelland\Levelland.lua:62: in function <Interface\AddOns\Levelland\Levelland.lua:56>
[C]: in function `GossipFrameActiveQuestsUpdate'
Interface\FrameXML\GossipFrame.lua:41: in function `GossipFrameUpdate'
Interface\FrameXML\GossipFrame.lua:33: in function <Interface\FrameXML\GossipFrame.lua:10>

Locals: 

  The count of that last error message seemed to increase without any limits, so I had to deactivate the addon after a while. I hope it helps... Thanks for your work anyway !

  Cheers !
